src/ol/interaction/modifyinteraction.js

```diff
@@ -959,6 +1044,12 @@ ol.interaction.Modify.prototype.insertVertex_ = function(segmentData, vertex) {
       coordinates = geometry.getCoordinates();
       coordinates[depth[0]].splice(index + 1, 0, vertex);
       break;
+    case ol.geom.GeometryType.RECTANGLE:
```

Comment:
Hmm, do you have any thoughts on the cleanest way to communicate the state (either dragging corner or dragging side) to the dragging context?